git_machete/client.py

```diff
@@ -1656,6 +1657,57 @@ def flush_caches(self) -> None:
         self.__branch_defs_by_sha_in_reflog = None
         self.__git.flush_caches()
 
+    def checkout_github_pr(self, pr_no: int) -> None:
+        org: str
+        repo: str
+        remote: str
+        remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
+        debug('checkout_github_pr', f'organization is {org}, repository is {repo}')
+
+        all_prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
```

Comment:
`GET /repos/.../.../pulls` is invoked 3 times in total, not good ☹️ this causes the entire operation to take like 5-10 seconds :/

That's what I've got from calling `git machete github checkout-pr 229 --debug`:

```
checkout_github_pr: organization is VirtusLab, repository is git-machete
__fire_github_api_request(GET, /repos/VirtusLab/git-machete/pulls, ..., ...): firing a GET request to https://api.github.com/repos/VirtusLab/git-machete/pulls with a bearer token and request body <none>
checkout_github_pr: found PR #229 by BartoszSiwinski: improve/parser -> develop
>>> git fetch origin
>>> git config --get remote.asford.url
<stdout>:
git@github.com:asford/git-machete.git

>>> git config --get remote.origin.url
<stdout>:
git@github.com:VirtusLab/git-machete.git

__fire_github_api_request(GET, /repos/VirtusLab/git-machete/pulls, ..., ...): firing a GET request to https://api.github.com/repos/VirtusLab/git-machete/pulls with a bearer token and request body <none>
>>> git config --get remote.asford.url
<stdout>:
git@github.com:asford/git-machete.git

>>> git config --get remote.origin.url
<stdout>:
git@github.com:VirtusLab/git-machete.git

__fire_github_api_request(GET, /user, ..., ...): firing a GET request to https://api.github.com/user with a bearer token and request body <none>
sync_annotations_to_github_prs(): Current GitHub user is PawelLipski
__fire_github_api_request(GET, /repos/VirtusLab/git-machete/pulls, ..., ...): firing a GET request to https://api.github.com/repos/VirtusLab/git-machete/pulls with a bearer token and request body <none>
sync_annotations_to_github_prs(): PR #230 by rifel123: bugfix/create-pr -> enchance/checkout-pr does NOT correspond to a managed branch
sync_annotations_to_github_prs(): PR #229 by BartoszSiwinski: improve/parser -> develop corresponds to a managed branch
sync_annotations_to_github_prs(): PR #214 by rifel123: testing/github_methods -> develop corresponds to a managed branch
sync_annotations_to_github_prs(): PR #205 by rifel123: enchance/checkout-pr -> testing/github_methods corresponds to a managed branch
>>> git checkout --quiet improve/parser --
Pull request #229 checked out at local branch improve/parser
```